internal/oidc/token/token_handler.go

```diff
@@ -187,6 +189,18 @@ func upstreamOIDCRefresh(ctx context.Context, session *psession.PinnipedSession,
 			WithDebugf("provider name: %q, provider type: %q", s.ProviderName, s.ProviderType))
 	}
 	if refreshedGroups != nil {
+		oldGroupsInterface := session.Fosite.Claims.Extra[oidc.DownstreamGroupsClaim].([]interface{})
+		oldGroups := []string{}
+		for _, i := range oldGroupsInterface {
+			oldGroups = append(oldGroups, i.(string))
```

Comment:
Can this `panic`?

Reply:
Good catch. I forgot to switch to the nice helper function that handles panics and such that I had written in the previous group change PR 🤦.